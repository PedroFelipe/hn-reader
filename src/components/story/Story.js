import React, { Component } from 'react'
import { isEmpty } from 'ramda';

import { client } from '../../index'
import { slice } from 'ramda'

import './story.css'

class Story extends Component {
  constructor(props) {
    super(props)
  }

  componentWillMount() {
    const { fetchStory, match } = this.props

    fetchStory(match.params.storyId)
  }

  componentDidUpdate(prevProps) {
    const { fetchStoryComments, match, story } = this.props

    if (isEmpty(prevProps.story.open) && !isEmpty(story.open)) {
      story.open.kids.forEach((commentId) => {
        fetchStoryComments(commentId)
      })
    }
  }

  render() {
    const { router } = this.props

    // console.log(this.props)

    return (
      <div className="story">
        <section className="section">

        </section>
      </div>
    )
  }
}

export default Story
