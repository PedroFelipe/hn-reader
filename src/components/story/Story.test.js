import React from 'react'
import renderer from 'react-test-renderer';

import Story from './Story'

test('Story renders correctly', () => {
  const match = { params: {} }
  const tree = renderer.create(
    <Story match={match} fetchStory={() => } fetchStoryComments={() => {}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
