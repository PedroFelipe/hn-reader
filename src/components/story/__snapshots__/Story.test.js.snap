// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story renders correctly 1`] = `
<div
  className="story"
>
  <section
    class="section"
  />
</div>
`;
